### PR TITLE
Sidebar: remove target=_blank from WP admin link

### PR DESCRIPTION
**Pull request by simison**
Removes `target=_blank` from the sidebar WP Admin link; 

- Customer's intent when pressing this button is not to visit another site (like support document) and come back to Calypso. They're leaving one admin dashboard for another. It's confusing to end up with two admin dashboard in two separate tabs.
- WP Admin should be seen as "just another page in their dashboard experience", just like Reader or Customizer.
- It's ok (albeit unfortunate) if the URL and UI are very different without new tabs-switch. So are iframed Customizer, iframed post-editor and the rest of Calypso kinda different looking, too.
- New tab breaks natural back/forward browser flow
- Tabs are kinda annoying on mobile.

#### Changes proposed in this Pull Request

* Remove `target=_blank` from the sidebar WP Admin link
* Remove `external` icon

#### Testing instructions
Have a Jetpack site connected (e.g. http://jurassic.ninja/?create)

Sidebar Before

Screenshot 2019-05-24 at 10 13 49

Sidebar After

Screenshot 2019-05-24 at 10 05 19

### CODE REVIEW

File: client/my-sites/sidebar/sidebar.jsx [hunk 1]
```diff
@@ -507,15 +507,9 @@ export class MySitesSidebar extends Component {
 		/* eslint-disable wpcalypso/jsx-classname-namespace*/
 		return (
 			<li className="wp-admin">
-				<a
-					onClick={ this.trackWpadminClick }
-					href={ adminUrl }
-					target="_blank"
-					rel="noopener noreferrer"
-				>
+				<a onClick={ this.trackWpadminClick } href={ adminUrl }>
```
Comment (sirreal) [inline client/my-sites/sidebar/sidebar.jsx]: We might use `ExternalLink` here, we don't need to have it open in another window:

https://github.com/Automattic/wp-calypso/blob/master/client/components/external-link/index.jsx

File: client/my-sites/sidebar/sidebar.jsx [hunk 2]
```diff
@@ -507,15 +507,9 @@ export class MySitesSidebar extends Component {
...
 					<Gridicon icon="my-sites" size={ 24 } />
 					<span className="menu-link-text">{ this.props.translate( 'WP Admin' ) }</span>
-					<Gridicon icon="external" size={ 24 } />
```
Comment (sirreal) [inline client/my-sites/sidebar/sidebar.jsx]: `ExternalLink` would add this, but I'm unclear as to whether this means "you're navigating outside of Calypso" or "we're going to open a new window."

I believe it's the former and we should keep the icon. `ExternalLink` will handle that.

**Comment by simison** [comment]
@sirreal Thanks! I added `ExternalLink` component. Good for another review! 

Note that the icon has 1px position difference between `master` and this because `ExternalLink` component adds is:
```css
.gridicon.needs-offset > use:first-child, .gridicon.needs-offset > g:first-child {
    transform: translate(1px, 1px);
}
```

Location change isn't noticeable in practise and the icon is crispier than it was before:
image

File: client/layout/sidebar/style.scss [hunk 3]
```diff
@@ -144,12 +144,13 @@
 
 		// External indicator for sections that aren't available in Calypso
 		&.gridicons-external {
+			height: 18px;
+			left: auto;
+			margin: 0;
 			position: absolute;
 			right: 15px;
-			left: auto;
+			top: auto;
```
Comment (simison) [inline client/layout/sidebar/style.scss]: Without `top:auto` CSS from `.external-link` component was leaking in otherwise. (`top: 2px`)

**Comment by simison** [comment]
Convo/question about this change in p1559299679020600-slack-calypso

**Pull request merged**